src/pyodide/internal/_workers.py

```diff
@@ -942,6 +984,39 @@ def wrapped_init(self, *args, **kwargs):
 
     cls.__init__ = wrapped_init
 
+    run_fn = getattr(cls, "on_run", None)
```

Comment:
Can you move the code you added here to a separate function and call it from here?